### PR TITLE
MultiChildLayoutDelegate should assert that all children have had layoutChild called exactly once

### LINKED ISSUE #692: MultiChildLayoutDelegate should assert that all children have had layoutChild called exactly once
...and should say which child you missed if you did miss one.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #692

### CODE REVIEW

File: packages/flutter/lib/src/rendering/custom_layout.dart [hunk 1]
```diff
@@ -47,11 +57,20 @@ abstract class MultiChildLayoutDelegate {
...
         assert(childParentData.id != null);
         _idToChild[childParentData.id] = child;
+        assert(() {
+          _debugChildrenNeedingLayout.add(child);
+          return true;
+        });
         child = childParentData.nextSibling;
       }
       performLayout(size, constraints);
+      assert(_debugChildrenNeedingLayout.isEmpty);
```
Comment (HansMuller) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: An explanatory message would really help here.

**Comment by HansMuller** [comment]
LGTM

File: packages/flutter/lib/src/rendering/custom_layout.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: Good idea.

Comment (Hixie) [inline packages/flutter/lib/src/rendering/custom_layout.dart]: Yeah, printing the layout IDs of the children that didn't get laid out would be spectacular.

**Pull request merged**